broker.go

```diff
@@ -1013,6 +1026,14 @@ func (b *Broker) sendInternal(rb protocolBody, promise *responsePromise) error {
 		return ErrUnsupportedVersion
 	}
 
+	// validate the request is using a supported API version
+	if apiVersions, ok := b.brokerAPIVersions[rb.key()]; ok {
+		if rb.version() < apiVersions.MinVersion || rb.version() > apiVersions.MaxVersion {
+			return fmt.Errorf("%w: unsupported API version %d for %T, supported versions are [%d-%d]",
```

Comment:
Proper interval notation would be `[%d,%d]`, though `%d-%d` is just as reasonable as a non–math-geek version.

Reply:
I'll keep it non-math-geeky then, for simplicity 🙂